README.md

```diff
@@ -1,2 +1,432 @@
+
+The general strategy for using one of the library templates in the parent microservice Helm chart is to create a template for the K8s object formateted as so:
+
+```
+{{- include "ffc-helm-library.secret" (list . "ffc-microservice.secret") -}}
+{{- define "ffc-microservice.secret" -}}
+# Microservice specific configuration in here
+{{- end -}}
+```
+
+This example would be for `template/secret.yaml` in the `ffc-microservice` Helm chart. The initial `include` statement can be wrapped in an `if` statement if you only wish to create the K8s object based on a condition. E.g., only create the secret is a `pr` value has been set in `values.yaml`:
+
+```
+{{- if .Values.pr }}
+{{- include "ffc-helm-library.secret" (list . "ffc-microservice.secret") -}}
+{{- end }}
+{{- define "ffc-microservice.secret" -}}
+# Microservice specific configuration in here
+{{- end -}}
+```
+
+
+### All template required values
+
+All the K8s object templates in the library require the following values to be set in the parent microservice Helm chart's `values.yaml`:
+
+```
+name: <string>
+namespace: <string>
+environment: <string>
+```
+
+### Cluster IP service template
+
+* Template file: `_cluster-ip-service.yaml`
+* Template name: `ffc-helm-library.cluster-ip-service`
+
+A K8s `Service` object of type `ClusterIP`.
+
+A basic usage of this object template would involve the creation of `templates/cluster-ip-service.yaml` in the parent Helm chart (e.g. `ffc-microservice`) containing:
+
+```
+{{- include "ffc-helm-library.cluster-ip-service" (list . "ffc-microservice.service") -}}
+{{- define "ffc-microservice.service" -}}
+# Microservice specific configuration in here
+{{- end -}}
+```
+
+#### Required values
+
+The following values need to be set in the parent chart's `values.yaml` in addition to the globally required values [listed above](#all-template-required-values):
+
+```
+container:
+  port: <integer>
+```
+
+### Container template
+
+* Template file: `_container.yaml`
+* Template name: `ffc-helm-library.container`
+
+A template for the container definition to be used within a K8s `Deployment` object.
+
+A basic usage of this object template would involve the creation of `templates/_container.yaml` in the parent Helm chart (e.g. `ffc-microservice`). Note the `_` in the name. This template is part of the `Deployment` object definition and will be used in conjunction the `_deployment.yaml` template ([see below](#deployment-template)). As a minimum, `templates/_container.yaml` would define environment variables and liveness/readiness probes, e.g.:
+
```

Comment:
Not all services will define liveness and readiness probes as they may not be applicable.  So maybe change the wording from `as a minimum`